kafka-client/src/main/java/dev/responsive/kafka/bootstrap/ChangelogMigrationTool.java

```diff
@@ -0,0 +1,124 @@
+import static org.apache.kafka.clients.consumer.ConsumerConfig.AUTO_OFFSET_RESET_CONFIG;
+import static org.apache.kafka.streams.StreamsConfig.REUSE_KTABLE_SOURCE_TOPICS;
+import static org.apache.kafka.streams.StreamsConfig.TOPOLOGY_OPTIMIZATION_CONFIG;
+
+import dev.responsive.kafka.api.ResponsiveKafkaStreams;
+import dev.responsive.kafka.api.config.ResponsiveMode;
+import dev.responsive.kafka.api.stores.ResponsiveKeyValueParams;
+import dev.responsive.kafka.api.stores.ResponsiveStores;
+import java.time.Instant;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.processor.api.Processor;
+import org.apache.kafka.streams.processor.api.ProcessorContext;
+import org.apache.kafka.streams.processor.api.Record;
+
+public class ChangelogMigrationTool {
+
+  private static final org.slf4j.Logger LOG =
+      org.slf4j.LoggerFactory.getLogger(ChangelogMigrationTool.class);
+
+  private final Consumer<Record<byte[], byte[]>> processor;
+  private final ChangelogMigrationConfig config;
+  private final Properties properties;
+  private final ResponsiveKeyValueParams params;
+
+  @SuppressWarnings("unused") // reason: public API
+  public ChangelogMigrationTool(
+      final Properties properties,
+      final ResponsiveKeyValueParams params
+  ) {
+    this(properties, params, r -> {});
+  }
+
+  // Visible for testing
+  ChangelogMigrationTool(
+      final Properties properties,
+      final ResponsiveKeyValueParams params,
+      final Consumer<Record<byte[], byte[]>> processor
+  ) {
+    this.processor = processor;
+
+    properties.put(TOPOLOGY_OPTIMIZATION_CONFIG, REUSE_KTABLE_SOURCE_TOPICS);
+    properties.put(AUTO_OFFSET_RESET_CONFIG, "earliest");
+    properties.put(RESPONSIVE_MODE, ResponsiveMode.MIGRATE.name());
+
+    // it is possible to push this number higher, but this is a relatively safe
+    // number and still gets decent performance -- use putIfAbsent in case the
+    // customer wants to override this to push more performance at the risk of
+    // stability
+    properties.putIfAbsent(STORE_FLUSH_RECORDS_TRIGGER_CONFIG, 10_000);
+
+    config = new ChangelogMigrationConfig(properties);
+    this.properties = properties;
+    this.params = params;
+  }
+
+  public ResponsiveKafkaStreams buildStreams() {
+    final StreamsBuilder builder = new StreamsBuilder();
+    final String source = config.getString(CHANGELOG_TOPIC_CONFIG);
+
+    final KTable<byte[], byte[]> table =
+        builder.table(
+            source,
+            Materialized
+                .<byte[], byte[]>as(ResponsiveStores.keyValueStore(params))
+                .withValueSerde(Serdes.ByteArray())
```

Comment:
These will set the serdes to use for reading/writing the store, but does it also get used for deserializing the source topic? Consider passing `Consumed` to `table` to be extra sure.